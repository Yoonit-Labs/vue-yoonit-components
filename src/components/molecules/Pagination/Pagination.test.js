/**
 * ██╗   ██╗ ██████╗  ██████╗ ███╗   ██╗██╗████████╗
 * ╚██╗ ██╔╝██╔═══██╗██╔═══██╗████╗  ██║██║╚══██╔══╝
 *  ╚████╔╝ ██║   ██║██║   ██║██╔██╗ ██║██║   ██║
 *   ╚██╔╝  ██║   ██║██║   ██║██║╚██╗██║██║   ██║
 *    ██║   ╚██████╔╝╚██████╔╝██║ ╚████║██║   ██║
 *    ╚═╝    ╚═════╝  ╚═════╝ ╚═╝  ╚═══╝╚═╝   ╚═╝
 *
 * https://yoonit.dev - about@yoonit.dev
 *
 * Vue Yoonit Components
 * VueJS Atomic Design System framework
 *
 * Vitória Costa, Tiago Brito, Fernando Junior, Sabrina Sampaio, Gabriel Mule, Gabriel Moraes, Gabriel Rizzo & Luigui Delyer @ 2020-2021
 */

import { mount } from '@vue/test-utils'
import yooPagination from './Pagination.vue'
import PropsConfig from './Pagination.config'

const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return mount(yooPagination, {
    slots: { default: SlotText },
    propsData: {
      totalPages: 5,
      currentPage: 0
    }
  })
}

describe('YooPagination Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.classes('yoo-pagination')).toBe(true)
    expect(wrapper.find('.yoo-stepper').exists()).toBe(true)
    expect(wrapper.find('.yoo__container--button').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('disableFirstAction', () => {
      it('Include disableFirstAction class: .yoo-btn--disabled', async () => {
        await wrapper.setProps({ disableFirstAction: true })
        expect(wrapper.find('.yoo-btn--disabled').exists()).toBe(true)
      })
    })
    describe('variationButtonNext', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.variationButtonNext.options.includes(yooPagination.props.variationButtonNext.default)).toBe(true)
      })
      PropsConfig.variationButtonNext.options.forEach(variation => {
        it('Include variation class: ', async () => {
          await wrapper.setProps({ variation })
          expect(wrapper.find(`.yoo-btn--${variation}`).exists()).toBe(true)
        })
      })
    })

    describe('counterType', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.counterType.options.includes(yooPagination.props.counterType.default)).toBe(true)
      })
      PropsConfig.counterType.options.forEach(counterType => {
        it(`Include counterType class: .yoo-stepper__${counterType}`, async () => {
          await wrapper.setProps({ counterType })
          expect(wrapper.find(`.yoo-stepper__${counterType}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props

  describe('Events', () => {
    describe('Click in Back Button', () => {
      it('Emits Click Back Button Event - Check First', async () => {
        await wrapper.find('.button__back').vm.$emit('onClick')
        expect(wrapper.emitted().tapPage[0]).toEqual(['first'])
      })
      it('Emits Click Back Button Event - Check Back', async () => {
        await wrapper.setProps({ currentPage: 2 })
        await wrapper.find('.button__back').vm.$emit('onClick')
        expect(wrapper.emitted().tapPage[0]).toEqual(['back'])
      })
    })

    describe('Click in Last Button', () => {
      it('Emits Click last Button Event', async () => {
        await wrapper.find('.button__last').vm.$emit('onClick')
        expect(wrapper.emitted()).toHaveProperty('tapPage')
      })
    })

    describe('Click in Next Button', () => {
      it('Emits Click Next Button Event - Check Next step', async () => {
        await wrapper.find('.button__next').vm.$emit('onClick')
        expect(wrapper.emitted().tapPage[0]).toEqual(['next'])
      })
      it('Emits Click Next Button Event - Check Last Step', async () => {
        await wrapper.setProps({ currentPage: 4 })
        await wrapper.find('.button__next').vm.$emit('onClick')
        expect(wrapper.emitted().tapPage[0]).toEqual(['last'])
      })
    })

    describe('Click', () => {
      it('Emits Click Event', async () => {
        await wrapper.find('.button__next--grow').vm.$emit('onClick')
        expect(wrapper.emitted()).toHaveProperty('tapPage')
      })
    })
  }) // describe Events
})
