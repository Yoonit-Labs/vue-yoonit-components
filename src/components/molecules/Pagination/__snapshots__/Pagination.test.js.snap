// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooPagination Component Matches Snapshot 1`] = `
<div class="yoo-flex-layout yoo-pagination" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-between; align-items: stretch; align-content: stretch;">
  <div class="yoo__container--button button__back"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--outline yoo-btn--normal yoo-btn--animation-none"><i class="yoo-btn__icon fas fa-arrow-left yoo-btn__icon--normal" style="color: rgb(255, 255, 255);"></i>
      <p class="yoo-btn__text"></p>
      <!---->
    </button></div>
  <div class="yoo-flex-layout yoo-stepper" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-around; align-items: center; align-content: stretch;">
    <!---->
    <div class="yoo-stepper__dot m__r--m m__l--m yoo-stepper__dot--active"></div>
    <div class="yoo-stepper__dot m__r--m m__l--m"></div>
    <div class="yoo-stepper__dot m__r--m m__l--m"></div>
    <div class="yoo-stepper__dot m__r--m m__l--m"></div>
    <div class="yoo-stepper__dot m__r--m m__l--m"></div>
  </div>
  <div class="yoo__container--button button__next"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--base yoo-btn--normal yoo-btn--animation-none"><i class="yoo-btn__icon fas fa-arrow-right yoo-btn__icon--normal" style="color: rgb(255, 255, 255);"></i>
      <p class="yoo-btn__text"></p>
      <!---->
    </button></div>
  <div class="yoo__container--button button__next--grow" flexgrow="1" style="display: none; flex-grow: 1;"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--base yoo-btn--normal yoo-btn--animation-none">
      <!---->
      <p class="yoo-btn__text"></p><i class="yoo-btn__icon fas fa-arrow-right yoo-btn__icon--normal" style="color: rgb(255, 255, 255);"></i>
    </button></div>
  <div class="yoo__container--button button__last" flexgrow="1" style="display: none; flex-grow: 1;"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--base yoo-btn--normal yoo-btn--animation-none">
      <!---->
      <p class="yoo-btn__text"></p><i class="yoo-btn__icon fas fa-arrow-right yoo-btn__icon--normal" style="color: rgb(255, 255, 255);"></i>
    </button></div>
</div>
`;
